src/main/java/frc/robot/commandGroups/CommandGroups.java

```diff
@@ -0,0 +1,93 @@
+package frc.robot.commandGroups;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.GripperState;
+import frc.robot.scoreStates.ScoreState;
+import frc.robot.subsystems.conveyor.Conveyor;
+import frc.robot.subsystems.elevator.Elevator;
+import frc.robot.subsystems.elevator.ElevatorConstants;
+import frc.robot.subsystems.gripper.Gripper;
+import frc.robot.subsystems.gripper.GripperConstants;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.shooter.Shooter;
+import frc.robot.subsystems.shooter.ShooterConstants;
+import java.util.function.Supplier;
+
+public class CommandGroups {
+    private static CommandGroups INSTANCE;
+    private final Intake intake;
+    private final Gripper gripper;
+    private final Elevator elevator;
+    private final Shooter shooter;
+    private final Conveyor conveyor;
+    private final GripperState gripperState;
+
+    private CommandGroups() {
+        intake = Intake.getInstance();
+        gripper = Gripper.getInstance();
+        elevator = Elevator.getInstance();
+        shooter = Shooter.getInstance();
+        conveyor = Conveyor.getInstance();
+        gripperState = new GripperState();
+    }
+
+    public static CommandGroups getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new CommandGroups();
+        }
+        return INSTANCE;
+    }
+
+    public Command elevatorGripperMinPosition() {
+        return Commands.parallel(
+                elevator.setHeight(ElevatorConstants.MIN_HEIGHT),
+                gripper.setWristPosition(GripperConstants.WRIST_FOLD_POSITION.mutableCopy()));
```

Comment:
1. Both commands don't have end conditions. In my opinion, it's much better to manage these inside the subsystem itself (will you ever want to set a subsystem's setpoint and end not when it reaches it?), but @vichik123 might disagree.
2. Be consistent: if `MIN_HEIGHT` is a `MutableMeasure` then `WRIST_FOLD_POSITION` should also be of this type. I just noticed that we should have just make everything receive `Measure` and not `MutableMeasure` and this would have prevented all the funny `.mutableCopy()`s everywhere. I don't know if it's worth changing now, @vichik123 will make the call.
3. These constants are used only in this class, so they should be stored in this package.

Reply:
c1ff82e
The elevator command does have end condition
8619a02
f653bcc
9715ee9

Reply:
Nice!
1. About c1ff82e, I think it's better to leave it a `runOnce()` with a `waitUntil` for the setpoint -- this better represents what you want.
3. Good job organizing these constants. Note that there are still constants which are used only in this class, but are stored outside of this package.

Reply:
cb386cf
1260dff

Reply:
1. Now there is no end condition at all 🙂 I didn't suggest to revert c1ff82e, but to simply refactor it a bit.

Reply:
There is no need for an end condition this is runOnce

Reply:
But now the command will end immediately, don't you want to wait until you actually reached the setpoint?

Reply:
652f84d